### PR TITLE
feat(tree): add mat tree node toggle component 

### PR DESCRIPTION
**Pull request by tinayuangao**

### CODE REVIEW

File: src/lib/tree/toggle.html [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+        [disabled]="!isExpandable"
+        [attr.aria-label]="ariaLabel">
+
+  <svg *ngIf="isExpandable && isExpanded"
+       fill="#000000" height="24" viewBox="0 0 24 24" width="24" xmlns="http://www.w3.org/2000/svg">
+    <path d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"/>
+    <path d="M0 0h24v24H0z" fill="none"/>
+  </svg>
+
+  <svg *ngIf="isExpandable && !isExpanded"
```
Comment (jelbourn) [inline src/lib/tree/toggle.html]: Instead of two SVGs, could we just apply CSS to transform one?

File: src/lib/tree/toggle.ts [hunk 2]
```diff
@@ -22,3 +22,43 @@ import {CdkTreeNodeToggle} from '@angular/cdk/tree';
...
+  },
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class MatTreeNodeToggleComponent<T> extends CdkTreeNodeToggle<T> {
+  /** Whether the toggle is recursive */
+  @Input('matTreeNodeToggleRecursive') recursive: boolean = false;
+
+  /** Used to set the aria-label attribute on the underlying input element. */
+  @Input('aria-label') ariaLabel: string | null = null;
```
Comment (jelbourn) [inline src/lib/tree/toggle.ts]: We should probably mention that this is needed in the docs

**Comment by tinayuangao** [comment]
@jelbourn Updated docs, css and svg. Please take a look. Thanks!

**Comment by jelbourn** [comment]
@tinayuangao just noticed the bazel build file needs an update

**Pull request closed without merging**